src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+type Color = 'red' | 'blue' | 'green';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
 
+function invalidValue(...args: number[]): boolean {
+  return args.some((param : number) => param <= 0);
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (invalidValue(this.a, this.b, this.c)) {
+      throw new Error('Invalid value');
+    }
+
+    const largest: number = Math.max(this.a, this.b, this.c);
 
-export class Circle {
+    if ((this.a + this.b + this.c - largest) <= largest) {
+      throw new Error('Sides 1, 2 and 3 can not form a triangle');
+    }
+  }
 
+  getArea(): number {
+    const semiPerimeter : number = (this.a + this.b + this.c) / 2;
```

Comment:
don't add type for primitive js types
```suggestion
    const semiPerimeter = (this.a + this.b + this.c) / 2;
```